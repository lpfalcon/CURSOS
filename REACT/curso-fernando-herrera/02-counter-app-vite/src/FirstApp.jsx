import PropTypes from "prop-types";
const FirstApp = ({ title, subTitle, name }) => {
  return (
    <>
      <h1 data-testid="test-title">{ title }</h1>
      <h3>{ subTitle }</h3>
      <h3>{ name }</h3>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Minus aperiam,
        rerum impedit repellendus delectus minima, eveniet quisquam eligendi
        tempore quaerat, deserunt tenetur fugiat doloremque nisi consequuntur
        pariatur quam adipisci facilis?
      </p>
    </>
  );
};

FirstApp.propTypes = {
  name: PropTypes.string,
  subTitle: PropTypes.string,
  title: PropTypes.string.isRequired,
};

FirstApp.defaultProps = {
  name: "No name",
  subTitle: "No subTitle",
  // title: "No title",
};
export default FirstApp;
