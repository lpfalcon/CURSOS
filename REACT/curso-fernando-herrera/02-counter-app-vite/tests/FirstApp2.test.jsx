import { render, screen } from "@testing-library/react"
import FirstApp from "../src/FirstApp"

describe('FirstApp test', () => {
    const title = 'Hello World'
    const subTitle = `I'm a subtitle`;
    test('Sholud Match with the snapshot', () => {

        const { container } = render(<FirstApp title={title} />);

        expect(container).toMatchSnapshot();
    })

    test('Should show a message "Hello Word"', () => {

        render(<FirstApp title={title} />);

        expect(screen.getByText(title)).toBeTruthy();

    })

    test('Should show a title in an H1', () => {

        render(<FirstApp title={title} />);

        expect(screen.getByRole('heading', { level: 1 }).innerHTML)
            .toContain(title);

    })

    test('Should show a subtitle sent in props', () => {

        render(<FirstApp 
            title={title} 
            subTitle={subTitle}
            />);

        expect(screen.getAllByText(title).length).toBeGreaterThan(0);
    })

})
