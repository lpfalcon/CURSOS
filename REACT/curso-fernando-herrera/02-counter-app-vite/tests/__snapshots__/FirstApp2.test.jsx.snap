// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FirstApp test Sholud Match with the snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hello World
  </h1>
  <h3>
    No subTitle
  </h3>
  <h3>
    No name
  </h3>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Minus aperiam, rerum impedit repellendus delectus minima, eveniet quisquam eligendi tempore quaerat, deserunt tenetur fugiat doloremque nisi consequuntur pariatur quam adipisci facilis?
  </p>
</div>
`;
